Move data from git repo into DVC

fmahnke opened the pull request:
- Add large file support using DVC.
- Add existing large files to DVC and change code to use the updated path.
- Remove the large files from the git repository.

fmahnke committed:
build: add DVC dependencies

Add these Python dependencies to support DVC:

- dvc (provides the dvc command line utility)
- dvc-s3 (provides support for S3-compatible remotes)

fmahnke committed:
data: initialize DVC

Initialize DVC in the top level directory. Existing large files will be moved
from the git repository to DVC. Going forward, large data files will be stored
in DVC instead of the git repository.

fmahnke committed:
data: stop tracking tests/data in git repository

All large files are now tracked in DVC; remove them from git.

fmahnke committed:
data: add existing tests/data

Add the existing files tracked by git in tests/data to DVC at data/ocr.

fmahnke committed:
data: use DreamHost DreamObjects S3-compatible DVC remote

fmahnke committed:
test: point data paths to DVC data directory

Change all tests to look for their data in the configured DVC data directory,
instead of in the git repo.

fmahnke committed:
doc: update README.md with DVC instructions

Add instructions to README.md on how to configure the DVC data path and check
out a working copy of its data.

fmahnke committed:
data: add IoG charity stream video

cbebry reviewed (changes requested):
Looks good for the most part. We've discussed outside of this PR that we should consider moving some of the data that is code test-specific to the git project, and keep the data that is strictly for training the model inside of DVC.

I know at one point during our discussion that the `$ mortar video || echo FAIL` didn't work as expected. Does this happen for you as well?

fmahnke commented:
Yes. I'll make that change, and move the data used in `test_detector.py` back to git.

fmahnke committed:
test: keep code-specific test resources in git

Subsequent commits remove large data from the git repository. Track small data
that is used for tests that exercise code. This allows unit tests to run even
when the large data set is unavailable.

fmahnke committed:
feat(video): make input path a required argument of video command

Make the video command more of a general utility. Instead of trying to guess
what path the user wants, take the input path as a required argument.

fmahnke commented:
- Added small data for tests that directly exercise code in `tests/data/ocr`. This makes unit testing available even when large file storage isn't checked out.
- Made the `mortar video` command take a required input directory argument, instead of trying to guess it.
